sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -138,6 +139,23 @@ class KotlinSensor(
         }
     }
 
+    private fun fileHasChanged(inputFile: InputFile, contentHashCache: ContentHashCache): Boolean{
+        if(contentHashCache.isEnabled()){
+            val cacheContentIsDifferent = contentHashCache.hasDifferentContentCached(inputFile)
+            if(cacheContentIsDifferent){
+                LOG.debug("Cache contained a different hash for file ${inputFile.filename()}")
+                contentHashCache.writeContentHash(inputFile)
+                return true
+            }
+            contentHashCache.copyFromPreviousAnalysis(inputFile)
```

Comment:
I feel like there is too much caching-specific logic going on in the KotlinSensor here. I think we can move some of it to the ContentHashCache, which will also allow us to simplify some code there.